streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -454,6 +456,41 @@ public void flush() {
         }
     }
 
+    public void flushCache() {
+        RuntimeException firstException = null;
+        // attempting to flush the stores
+        if (!stores.isEmpty()) {
+            log.debug("Flushing all store caches registered in the state manager: {}", stores);
+            for (final StateStoreMetadata metadata : stores.values()) {
+                final StateStore store = metadata.stateStore;
+
+                try {
+                    // buffer should be flushed to send all records to changelog
+                    if (store instanceof TimeOrderedKeyValueBuffer) {
```

Comment:
Admit this is kinda hacky, but I'd have to do this for cached store and suppression buffer. Moving forward I think the first can be removed when we decouple caching with emitting, but for suppression buffer maybe we can have a more general way to fix it, for example maybe we could just have changelogger to always buffer itself so that suppression buffers do not need to buffer itself to changelogger. cc @vvcephei

Reply:
WDYT about adding a generic `FlushingRequiredStore` marker interface that all caching state stores and the suppression buffer would both implement. It seems weird to handle them separately.

We could even make this public and allow user custom state stores to implement this, but that might be opening a can of worms we will greatly regret 😉

Reply:
My plan is actually to remove the `flushCache` once we decoupled caching with emitting (see the TODO comment on the caller).

Reply:
So you're saying we'd still need to flush the suppression buffer but not the cache once we decouple caching from emitting? Or that we can remove this `flushCache` method altogether once that is done? Or that it will still do some flushing, but will not resemble the current `flushCache` method at all

Reply:
I'm thinking we can remove the whole `flushCache` method.

Reply:
Is there a ticket for that?

Reply:
I'm going to create the ticket of that very soon.